app/lib/features/space/settings/pages/visibility_accessibility_page.dart

```diff
@@ -0,0 +1,310 @@
+      ),
+    );
+  }
+
+  Widget spaceItemUI(Space? space, SpaceItem spaceItem) {
+    final avatar = ActerAvatar(
+      options: AvatarOptions(
+        AvatarInfo(
+          uniqueId: spaceItem.roomId,
+          displayName: spaceItem.spaceProfileData.displayName,
+          avatar: spaceItem.spaceProfileData.getAvatarImage(),
+        ),
+        size: 45,
+        badgesSize: 45 / 2,
+      ),
+    );
+
+    return Card(
+      margin: const EdgeInsets.symmetric(vertical: 5),
+      shape: RoundedRectangleBorder(
+        side: BorderSide(
+          color: Theme.of(context).colorScheme.primary,
+          width: 1.5,
+        ),
+        borderRadius: BorderRadius.circular(6),
+      ),
+      child: ListTile(
+        title: Text(
+          spaceItem.spaceProfileData.displayName ?? spaceItem.roomId,
+        ),
+        leading: avatar,
+        trailing: IconButton(
+          onPressed: () => removeSpace(space, spaceItem),
+          icon: Icon(
+            Atlas.trash,
+            color: Theme.of(context).colorScheme.error,
+          ),
+        ),
+      ),
+    );
+  }
+
+  Future<void> removeSpace(Space? space, SpaceItem spaceItem) async {
+    final room =
+        await ref.read(maybeRoomProvider(space!.getRoomIdStr()).future);
+    if (room != null) {
+      if (await space.isChildSpaceOf(spaceItem.roomId)) {
+        await room.removeParentRoom(spaceItem.roomId, null);
+      } else {
+        await space.removeChildRoom(spaceItem.roomId, null);
+      }
+    }
+    setState(() {
+      spaceList.remove(spaceItem);
+      if (spaceList.isEmpty) {
+        selectSpace(space.getRoomIdStr());
+      }
+    });
+  }
+
+  Future<void> selectSpace(String roomId) async {
+    try {
+      final spaceId = await selectSpaceDrawer(context: context);
+      if (spaceId != null) {
+        final spaceItem =
+            await ref.watch(maybeSpaceInfoProvider(spaceId).future);
+        final isAlreadyAdded =
+            spaceList.any((element) => element.roomId == spaceItem?.roomId);
+        if (spaceItem != null && !isAlreadyAdded) {
+          spaceList.add(spaceItem);
+          updateSpaceVisibility(
+            RoomVisibility.SpaceVisible,
+            spaceId,
+            roomId: roomId,
+          );
+          setState(() {});
+        }
+      }
+    } catch (e, st) {
+      _log.severe('Select Space Error =>>', e, st);
+      if (!mounted) return;
+      EasyLoading.showToast(
+        L10n.of(context).failedToLoadSpace(e),
+        toastPosition: EasyLoadingToastPosition.bottom,
+      );
+    }
+  }
+
+  Future<void> updateSpaceVisibility(
+    RoomVisibility value,
+    String spaceId, {
+    String? roomId,
+  }) async {
+    try {
+      EasyLoading.show(status: 'Updating space settings', dismissOnTap: false);
+      final sdk = await ref.read(sdkProvider.future);
+      final room = await ref.read(maybeRoomProvider(spaceId).future);
+      final space = await ref.read(maybeSpaceProvider(spaceId).future);
```

Comment:
See below for why these are needed:
```suggestion
  Future<void> updateSpaceVisibility(
    RoomVisibility value, {
    List<String>? newSpaceIds,
  }) async {
    try {
      EasyLoading.show(status: 'Updating space settings', dismissOnTap: false);
      final sdk = await ref.read(sdkProvider.future);
```